### PR TITLE
Output journal logs to CLI

### LINKED ISSUE #248: Display progress at cli
Connect the existing cli to the new internal state.

We want to ensure:
* validation errors are displayed with ~file/line/column numbers~
* other errors are caught and displayed
* the result of a deploy is displayed (whether success or failure)

## TODO

- [x] add a mode for adding a human readable text line to debug for each message processed by the journal
- [ ] check error display for:
  - [ ] validation errors
  - [ ] reconciliation errors
  - [ ] execution reverts
  - [ ] other errors?

### PR DESCRIPTION
**Pull request by zoeyTM**
- adds a `--logs` flag to `npx hardhat deploy2 ...` that enables logging journal write operations to the CLI

fixes #248

### CODE REVIEW

**Comment by zoeyTM** [comment]
Full output of successful deploy of every future type

Screen Shot 2023-07-19 at 11 02 05 PM

File: packages/core/src/new-api/internal/journal/utils/log.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    return console.log(
+      `execution failure - futureId: '${message.futureId}' error: '${message.error.message}'`
+    );
+  }
+
+  if (isWipeMessage(message)) {
+    return console.log(`wiping journal`);
+  }
+
+  throw new Error("Unknown journal message");
```
Comment (kanej) [inline packages/core/src/new-api/internal/journal/utils/log.ts:199]: Lets wrap this in an assertNever() style function, to get a compile time check when we add new Journalable messages.

File: packages/core/src/new-api/internal/journal/utils/log.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+export function logJournalableMessage(message: JournalableMessage): void {
+  /* start messages */
+
+  if (isDeployContractStartMessage(message)) {
+    return console.log(`deploy contract start - id: '${message.futureId}'`);
+  }
+
+  if (isCallFunctionStartMessage(message)) {
+    return console.log(`call function start - id: '${message.futureId}'`);
```
Comment (kanej) [inline packages/core/src/new-api/internal/journal/utils/log.ts:44]: Lets add the execution id for those messages that have it.

**Comment by alcuadrado** [comment]
looks good

**Pull request merged**